gas/config/tc-riscv.c

```diff
@@ -3707,6 +3712,18 @@ riscv_ip (char *str, struct riscv_cl_insn *ip, expressionS *imm_expr,
 		  ip->insn_opcode |= ENCODE_CV_BITMANIP_UIMM2 (imm_expr->X_add_number);
 		  ++oparg;
 		}
+		  else if (oparg[1] == '8')
+		  // b8: uimm6 bits unsigned immediate bits
+		{
+		  my_getExpression (imm_expr, asarg);
+		  check_absolute_expr (ip, imm_expr, FALSE);
+		  asarg = expr_end;
+		  if (imm_expr->X_add_number<0 || imm_expr->X_add_number>63)
+		  as_bad(_("immediate value must be 6-bit unsigned, %ld is out of range"),
+		  imm_expr->X_add_number);
+		  ip->insn_opcode |= ENCODE_CV_SIMD_UIMM6 (imm_expr->X_add_number);
+		  ++oparg;
+		}
```

Comment:
Not an error, I'm just curious why this is named `b8`.

Reply:
> Not an error, I'm just curious why this is named `b8`.

The prefix `b` has been assigned for CORE-V related immediates. This was defined when CV32E40Pv1 was implemented. I used 'b8' as this is the eighth immediate defined for CORE-V.

Reply:
Thanks, this makes sense.